apps/shell/app.js

```diff
@@ -141,10 +164,13 @@ server.on('upgrade', function upgrade(request, socket, head) {
       client_origin = request.headers['origin'],
       server_origin = custom_server_origin(default_server_origin(request.headers));
 
+  const host_path_rx = '/ssh/([^\\/\\?]+)([^\\?]+)?(\\?.*)?$';
+  const match = request.url.match(host_path_rx);
+  const host_in_whitelist = whitelist.includes(match[1]);
```

Comment:
Include these declarations with the others above (all in the same expression, comma delimited). It is good form to have the consistency. Everything should be safely declared as const.